### ISSUE TITLE
Auf CCU abgelernte Geräte bringen u.U. HomeKit durcheinander 

### ISSUE THREAD

**Issue by dehsgr** [rdmtc/RedMatic-HomeKit#122]
Ich habe heute ein paar Cuxd-Geräte, die nicht über RedMatic in HomeKit gepublished waren, von RaspberryMatic abgelernt, da nicht mehr benötigt. Cuxd war generell im HomeKit-Node nicht angehakt. Nach Reboot der CCU waren plötzlich alle via Config-Node gepublishten Geräte weg. Nur die mit der selben Bridge veröffentlichen Geräte (z.B. Systemvariablen) waren noch vorhanden. 

Während aller Versuche, das System wieder zum laufen zu bringen, stellte ich fest, dass es irgendwie mit der CCU-Config des Nodes (war bei mir localhost) im Zusammenhang zu stehen schien. Nachdem ich sie wieder auf die Vorgeschlagene CCU eingestellt habe (reguläre IP und automatisch detektierte Name) scheint die Bridge nun auch wieder einen Reboot zu überleben, ohne dass wieder Geräte verschwinden.

Im Config-Node waren sie noch korrekt vorhanden, aber wurden nur noch mit ihrer Seriennummer gelistet, anstatt mit Namen. Auch das Erstellen einer einer weiteren Bridge basierend auf der originalen CCU-Config zeigte dabei ein merkwürdiges Phänomen: das Laden der Aktoren lief in Endlosschleife.

Kann das Ganze ggf. mit dem letzten RM-Update im Zusammenhang stehen?

**Comment by hobbyquaker** [comment]
hattest/hast Du mehrere CCU config nodes? Falls ja: das sollte man nicht machen (ausser man hat mehrere CCUs). Wenn Du Redmatic einsetzt sollte es ausschliesslich den „localhost“ CCU config node geben.

**Comment by dehsgr** [comment]
Nein, hatte ich natürlich nicht. 

Den localhost hatte ich vor dem Problem auch – jetzt nicht mehr. Nun zeigt er auf die reguläre Netzwerk-IP und hat eine kryptische Hex-Folge als Namen.

**Comment by dehsgr** [comment]
Meine aktuelle CCU-Config sieht so aus:

<img width="490" alt="bildschirmfoto 2019-02-10 um 20 02 13" src="https://user-images.githubusercontent.com/4125820/52538103-cff36c80-2d6e-11e9-84fa-564d65a4135e.png">

Wenn Du weitere Infos benötigst, nur zu. ;-)

**Comment by hobbyquaker** [comment]
setz mal die `CCU Address` wieder auf `127.0.0.1` oder `localhost`. Tut jetzt eigentlich alles wieder soweit oder besteht aktuell noch ein Problem?

**Comment by dehsgr** [comment]
Mit der 192.168.2.4 läufts, mit localhost wieder die alten Probleme. Ich kann jetzt erst einmal damit leben. Schaun wir mal, ob es nochmal zu Problemen kommt.

**Comment by hobbyquaker** [comment]
hmm... schräg. Wenn Du magst schick mir mal deine flows.json an hobbyquaker@gmail.com dann schau ich mir das an.

**Comment by dehsgr** [comment]
Mache ich gerne, wenn Du mir verrätst, wie ich gespeicherte Credentials ausschließen kann. Die würde ich nämlich ungern übermitteln, da ich noch etliche andere Dienste integriert habe, weil Du es mit Deiner hervorragenden Arbeit ermöglicht hast, dass ich meine Homebridge beerdigen konnte. ;-)

...oder muss ich die händisch suchen und ersetzen?

**Comment by dehsgr** [comment]
...achso: RaspberryMatic und RedMatic inkl. aller Module sind übrigens auf aktuellem Patch-Level.

**Comment by hobbyquaker** [comment]
Normalerweise stecken credentials nicht mit in der flows.json, die werden in einer extra Datei gespeichert die zudem verschlüsselt ist. Außer der Entwickler des Nodes das Credentials speichert hat einen Fehler gemacht ;)

**Comment by dehsgr** [comment]
Ja, das habe ich gemerkt. An einigen Stellen ließ es sich (noch) nicht vermeiden. Geixxxte Mail ist Dir zugegangen.

**Comment by dehsgr** [comment]
Scheint durch #188 gefixt zu sein.

**Comment by hobbyquaker** [comment]
RedMatic-HomeKit **4.0.0** ist seit heute veröffentlicht – Installation in RedMatic 9 über den Node-RED-Palettenmanager (Suche nach `redmatic-homekit`), Release-Notes: https://github.com/rdmtc/RedMatic-HomeKit/releases/tag/v4.0.0.

Laut letztem Kommentar durch #188 behoben; in 4.0.0 sind die Accessory-Identitäten außerdem unabhängig von der ccu-connection-Konfiguration (UUIDs aus den CCU-Adressen).

Falls es mit 4.0.0 nicht wie erwartet funktioniert: bitte hier melden (Gerätetyp, Node-RED-Log mit Level „debug“ rund um den Start), dann wird das Issue wieder geöffnet.

---
*Written by Claude Fable on behalf of hobbyquaker.*
